pkg/upgrade/upgrade.go

```diff
@@ -0,0 +1,246 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/mitchellh/go-ps"
+	"github.com/opencontainers/selinux/go-selinux"
+	"github.com/rancher/k3s-upgrade/pkg/types"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func fileExists(path string) bool {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+func GetProcessPID() (int, error) {
+	processes, err := ps.Processes()
+	if err != nil {
+		return 0, err
+	}
+	for _, proc := range processes {
+		if strings.HasSuffix(proc.Executable(), "k3s") {
```

Comment:
@MonzElmasry when I suggested we use `strings.HasSuffix()` I didn't realize that `proc.Executable` was likely the full `/proc/${pid}/cmdline` which means this won't work as-is, I think.